Source/Core/Core/IOS/WFS/WFSI.cpp

```diff
@@ -257,20 +267,31 @@ IPCCommandResult WFSI::IOCtl(const IOCtlRequest& request)
   case IOCTL_WFSI_FINALIZE_TITLE_INSTALL:
   {
     std::string tmd_path;
-    if (m_patch_type == NOT_A_PATCH)
+
+    switch (m_patch_type)
     {
-      std::string title_install_dir = StringFromFormat("/vol/%s/_install/%s", m_device_name.c_str(),
-                                                       m_import_title_id_str.c_str());
-      std::string title_final_dir =
-          StringFromFormat("/vol/%s/title/%s/%s", m_device_name.c_str(),
-                           m_import_group_id_str.c_str(), m_import_title_id_str.c_str());
-      File::Rename(WFS::NativePath(title_install_dir), WFS::NativePath(title_final_dir));
+    case PATCH_TYPE_2:
+    {
+      // Delete content's default.dol
+      const std::string title_content_dir =
+          StringFromFormat("/vol/%s/title/%s/%s/content/", m_device_name.c_str(),
+                           m_current_group_id_str.c_str(), m_current_title_id_str.c_str());
+      File::Delete(WFS::NativePath(title_content_dir + "default.dol"));
 
-      tmd_path = StringFromFormat("/vol/%s/title/%s/%s/meta/%016" PRIx64 ".tmd",
-                                  m_device_name.c_str(), m_import_group_id_str.c_str(),
-                                  m_import_title_id_str.c_str(), m_import_title_id);
+      // Copy content's _default.dol to patch's directory
+      const std::string patch_dir =
+          StringFromFormat("/vol/%s/title/%s/%s/_patch/", m_device_name.c_str(),
+                           m_current_group_id_str.c_str(), m_current_title_id_str.c_str());
+      const std::string patch_content_dir = patch_dir + "content/";
+      File::CreateDir(WFS::NativePath(patch_dir));
+      File::CreateDir(WFS::NativePath(patch_content_dir));
+      File::Rename(WFS::NativePath(title_content_dir + "_default.dol"),
+                   WFS::NativePath(patch_content_dir + "default.dol"));
+
+      FinalizePatchInstall();
+      // Fall through intended.
```

Comment:
Could you place `[[fallthrough]]` here? Or `// [[fallthrough]]` in case we _still_ don't fully support C++17 everywhere...
Thats what most of the other places do, to be ready for more recent compilers.

Reply:
Done.